mlflow/transformers.py

```diff
@@ -0,0 +1,851 @@
+
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    path = pathlib.Path(path).absolute()
+
+    _validate_and_prepare_target_save_path(str(path))
+
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, str(path))
+
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, str(path))
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    resolved_task = _get_or_infer_task_type(transformers_model, task)
+
+    if not isinstance(transformers_model, transformers.Pipeline):
+        built_pipeline = _build_pipeline_from_model_input(transformers_model, resolved_task)
+    else:
+        built_pipeline = transformers_model
+
+    flavor_conf = _generate_base_flavor_configuration(built_pipeline, resolved_task)
+
+    components = _record_pipeline_components(built_pipeline)
+
+    if components:
+        flavor_conf.update(**components)
+
+    if processor:
+        warnings.warn(
+            "The model being saved contains a Processor component. Processors are not "
+            "compatible or available to be used if loading as pyfunc or used in a "
+            "spark_udf."
+        )
+        flavor_conf.update({_PROCESSOR_TYPE_KEY: _get_instance_type(processor, False)})
+
+    # Save the pipeline object
+    built_pipeline.save_pretrained(save_directory=path.joinpath(_PIPELINE_BINARY_FILE_NAME))
+
+    # Save the components explicitly to the components directory
+    _save_components(
+        root_path=path.joinpath(_COMPONENTS_BINARY_KEY),
+        component_config=components,
+        pipeline=built_pipeline,
+        processor=processor,
+    )
+
+    # Get the model card from either the argument or the HuggingFace marketplace
+    card_data = model_card if model_card is not None else _fetch_model_card(transformers_model)
+
+    # If the card data can be acquired, save the text and the data separately
+    if card_data:
+        write_to(str(path.joinpath(_CARD_TEXT_FILE_NAME)), card_data.text)
+        with path.joinpath(_CARD_DATA_FILE_NAME).open("w") as file:
+            yaml.safe_dump(card_data.data.to_dict(), stream=file, default_flow_style=False)
+
+    model_bin_kwargs = {_PIPELINE_BINARY_KEY: _PIPELINE_BINARY_FILE_NAME}
+
+    # Only allow a subset of task types to have a pyfunc definition.
+    # Currently supported types are NLP-based language tasks which have a pipeline definition
+    # consisting exclusively of a Model and a Tokenizer.
+    if _should_add_pyfunc_to_model(built_pipeline):
+        pyfunc.add_to_model(
+            mlflow_model,
+            loader_module="mlflow.transformers",
+            conda_env=_CONDA_ENV_FILE_NAME,
+            python_env=_PYTHON_ENV_FILE_NAME,
+            code=code_dir_subpath,
+            **model_bin_kwargs,
+        )
```

Comment:
Can we print a warning when the pyfunc flavor isn't added to the model? We do this for sklearn here: https://github.com/mlflow/mlflow/blob/2d87553ed073448a4a3564eea38c3f216ff7bbc7/mlflow/sklearn/__init__.py#L272-L285

Reply:
Added :)